client/elasticsearch/v2/client.go

```
@@ -151,6 +153,57 @@ func (cli *Client) AddDocument(ctx context.Context, indexName, documentID string
 	return nil
 }
 
+// DeleteDocumentByID deletes a document from the given index using the document ID.
+func (cli *Client) DeleteDocumentByID(ctx context.Context, index, documentID string) error {
+	documentType := DocumentType
+	documentPath := fmt.Sprintf("%s/%s/%s/%s", cli.URL, index, documentType, documentID)
+
+	_, status, err := cli.callElastic(ctx, documentPath, "DELETE", nil)
+	if err != nil {
+		return esError.StatusError{
+			Err:  fmt.Errorf("failed to delete document with ID '%s': %w", documentID, err),
+			Code: status,
+		}
+	}
+
+	if status != http.StatusOK && status != http.StatusAccepted && status != http.StatusNotFound {
+		return esError.StatusError{
+			Err:  fmt.Errorf("unexpected status code %d while deleting document with ID '%s'", status, documentID),
+			Code: status,
+		}
+	}
+
+	return nil
```

No need to implement this functionality in the v2 client as we don't use ES2 any more, I'd simply throw an error if this is called, similar to other functions in this package.
```
return errors.New("DeleteDocument is not supported for legacy client")
```